gorm-rest/src/main/groovy/yakworks/rest/gorm/responder/SecurityEntityResponderValidator.groovy

```diff
@@ -0,0 +1,49 @@
+/*
+* Copyright 2024 Yak.Works - Licensed under the Apache License, Version 2.0 (the "License")
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+package yakworks.rest.gorm.responder
+
+import groovy.transform.CompileStatic
+
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.core.Ordered
+
+import gorm.tools.hibernate.QueryConfig
+import gorm.tools.mango.api.QueryArgs
+import yakworks.security.gorm.UserSecurityConfig
+import yakworks.security.user.CurrentUser
+
+/**
+ * Sets user specific extended value for query timeout and max, if configured for logged in user.
+ */
+@CompileStatic
+class SecurityEntityResponderValidator implements EntityResponderValidator, Ordered {
+
+    @Autowired QueryConfig queryConfig
+    @Autowired UserSecurityConfig userSecurityConfig
+    @Autowired CurrentUser currentUser
+
+    @Override
+    QueryArgs validate(QueryArgs qargs) {
+
+        Integer timeout = userSecurityConfig.getQueryTimeout(currentUser)
+        if(!timeout) timeout = queryConfig.timeout
+        if(timeout) qargs.timeout(timeout)
+
+        Integer max = userSecurityConfig.getMax(currentUser)
```

Comment:
we should be blowing error if they are passsing in max more than they are allowed

Reply:
Fixed